src/Engine/babylon.engine.ts

```diff
@@ -5905,8 +5920,32 @@
             }
 
             // Upload data if needed. The texture won't be ready until then.
-            if (data) {
-                this.updateRawCubeTexture(texture, data, format, type, invertY, compression);
+            if (data && data.length) {
+                if (rgbd) {
+                    EnvironmentTextureTools.UploadLevelsAsync(texture, data as ArrayBufferView[][]).then(() => {
+                        texture.isReady = true;
+
+                        texture.onLoadedObservable.notifyObservers(texture);
+                        texture.onLoadedObservable.clear();
+
+                        if (onLoad) {
+                            onLoad();
+                        }
+                    }, reason => {
+                        if (onError) {
+                            onError("Failed to upload raw cube texture data to the GPU", reason);
+                        }
+                    });
+                }
+                else {
+                    Tools.SetImmediate(() => {

```

Comment:
why setImmediate ?

Reply:
I was originally going to use the onLoadedObservable to get notified when the texture is ready which would require updateRawCubeTexture to not fire the onLoadedObservable until after one frame to allow adding a callback in time. But I've added a onLoad callback just like createCubeTexture, so I don't need to do this anymore.

Reply:
Perfect :)